example/metro.config.js

```diff
@@ -1,49 +1,20 @@
-const { getDefaultConfig, mergeConfig } = require('@react-native/metro-config');
-const path = require('path');
-const escape = require('escape-string-regexp');
-const exclusionList = require('metro-config/src/defaults/exclusionList');
-const pak = require('../package.json');
+const path = require('node:path');
```

Comment:
Are these changes needed or leftover from example changes?

Reply:
The default `metro.config.js` that comes with React Native iOS/Android apps, and the modified one currently on `main`, does not support out-of-tree platforms. A few changes are needed to do things like alias `react-native` to `react-native-macos` on macOS. Additional changes would be needed if supporting `react-native-windows` in future.

One could express all that without `@rnx-kit/metro-config`, but it's not pretty. Here's the `metro.config.js` that you'd get if you set up a brand new React Native v0.75 project with both macOS and Windows support:

```js
const {getDefaultConfig, mergeConfig} = require('@react-native/metro-config');

const fs = require('fs');
const path = require('path');
const exclusionList = require('metro-config/src/defaults/exclusionList');

const rnwPath = fs.realpathSync(
  path.resolve(require.resolve('react-native-windows/package.json'), '..'),
);

//

/**
 * Metro configuration
 * https://facebook.github.io/metro/docs/configuration
 *
 * @type {import('metro-config').MetroConfig}
 */

const config = {
  //
  resolver: {
    blockList: exclusionList([
      // This stops "react-native run-windows" from causing the metro server to crash if its already running
      new RegExp(
        `${path.resolve(__dirname, 'windows').replace(/[/\\]/g, '/')}.*`,
      ),
      // This prevents "react-native run-windows" from hitting: EBUSY: resource busy or locked, open msbuild.ProjectImports.zip or other files produced by msbuild
      new RegExp(`${rnwPath}/build/.*`),
      new RegExp(`${rnwPath}/target/.*`),
      /.*\.ProjectImports\.zip/,
    ]),
    //
  },
  transformer: {
    getTransformOptions: async () => ({
      transform: {
        experimentalImportSupport: false,
        inlineRequires: true,
      },
    }),
  },
};

module.exports = mergeConfig(getDefaultConfig(__dirname), config);
```

[@rnx-kit/metro-config](https://microsoft.github.io/rnx-kit/docs/tools/metro-config) does all that under the hood, as well giving some other goodies like better monorepo support and supporting extending Expo's metro config.